### PR TITLE
Add batch and bash files, move project files under src folder. Change…

### PR DESCRIPTION
**Pull request by MariwanJ**
I wanted to have a correct way to build. And I also wanted to have project files under /src collected. 
To make it easier, I also changed the cmake file to have variable for easy project name change. 
There are two scripts that will make it easier to build for both linux and windows. Only thing that needs to be changed is the target name. hope this help others to use mbed-ce

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,10 +19,17 @@ include(mbed-os/tools/cmake/app.cmake) # Load Mbed CE toolchain file and basic b
...
 
-project(MbedCEHelloWorld) # TODO: change this to your project name
+#Keep files inside /src directory
+file(GLOB_RECURSE SOURCES
+     src/*.cpp
+     src/*.h
+)
+
+#Project name is based on the previous definition
+project(MbedCEHelloWorld)
```
Comment (JohnK1987) [inline CMakeLists.txt:29]: Shouldn't this line look like this?
```
project(MBED_APP_JSON_PATH)
```

**Comment by JohnK1987** [comment]
Hello,
I am glad you want to use MbedCE and do a contribution.
Unfortunately i am not sure this is the way where we want to go, i have no problem with Eclipse support but has to be little bit differently.

You made an edit of `mbed-ce-hello-world` according to your requirements but this is in collision with the rest of already implemented structure of Mbed project which is 10years old. For example like this.
```
#Keep files inside /src directory
file(GLOB_RECURSE SOURCES
     src/*.cpp
     src/*.h
)

add_executable(${PROJECT_NAME} ${SOURCES})
```
Mbed projects do not use **src** folder (i understand it is a standart in some IDEs). We can discuss about this but i believe you can achieve same via simple edit of top lvl cmka list like this for now.
```
add_executable(${PROJECT_NAME} scr/main.cpp)
```

Also files specific only to Eclipse or guide about *How run Eclipse* directly in the example which is for everyone makes no sense to me.

I would rather see a similar wiki page to [VS-Code](https://github.com/mbed-ce/mbed-os/wiki/Project-Setup:-VS-Code) but for Eclipse. This is correct place where we can put necessary steps about how to achieve this step by step.

BR, Jan

**Comment by MariwanJ** [comment]
Those configurations have nothing to do with Eclipse. It is just to organize the files. 
The thing with the Eclipse is ONLY the pdf file. So please do not mix.
I do not understand what you men by this statement  
> Mbed projects do not use src folder (i understand it is a standart in some IDEs). We can discuss about this but i believe you can achieve same via simple edit of top lvl cmka list like this for now.

The idea about this PR is to have both **batch and bash** files. Later I added the **/src** to put the project files inside a folder.  I have so many project that I collect project files inside a separate folder which is called src. Just to keep the root folder cleaner. mbed-os folder is not under /src.
And last I wanted to share my experience with eclipse. 
VSCode or Mbedstudio none of them was good for me to use it with mbed. It was impossible with mbedstudio to debug. I am not a fan of VScode either for cpp projects.
BR

**Comment by JohnK1987** [comment]
>  I wanted to share my experience with eclipse.

I appreciate this.

For the rest. Ok, my bad, i do not have experiences with Eclipse but then it is more easy. The batch/bash files are good for command line solution, for the rest it is just a mess what can be deleted and this can make confusing for newices.

From my point of view the hello world example should stay same. The requested changes is possible to do just in your top level cmake according to your own requirements.
More important is correct documentation about how to start with Eclipse. I already prepared something (inspired by your guide) - [Project-Setup:-Eclipse](https://github.com/mbed-ce/mbed-os/wiki/Project-Setup:-Eclipse).

**Comment by MariwanJ** [comment]
np

**Comment by multiplemonomials** [comment]
Just wanted to say, thanks for putting this together, it's interesting to see how this project can be adapted for Eclipse!  However, at this time I would prefer not to change the project structure for Eclipse support since Eclipse isn't one of our officially supported IDEs.

However!  You are welcome to add your Eclipse instructions to the [Mbed CE wiki](https://github.com/mbed-ce/mbed-os/wiki)!  Also, I think it would be pretty interesting if you could check out this Eclipse plugin: https://github.com/15knots/cmake4eclipse?tab=readme-ov-file.  It looks like it might make the process of setting up CMake projects like this one in Eclipse a lot easier!

**Comment by MariwanJ** [comment]
There is an issue with the cmake plugin. I couldn't get the configuration page of cmake inside eclipse working even with the latest eclipse. So for now, that plug in is not working for "me". Not sure if it is just my computer/setup. 
I updated my version (readme) to tell about what that version include. Would be good if I could put that line of batch file inside the cmake itself. but for know I just leave it like that. 
Thanks again, and hope you will have energy to keep working on the project :)

**Comment by MariwanJ** [comment]
Coming back to the discussion. Now, this line inside the cmake, doesn't bring all include libraries. 
`add_subdirectory(mbed-os) # Load Mbed OS build targets.  Must be added before any other sub directories`

I have in a project these lines .. they are failing. 
`
#include "EthernetInterface.h"
#include "TCPSocket.h"
`
Sure, I have the files all under /src. 
Now, since this is not a real make file, I don't really know what is going on using mbed tools. Don't really konw how to fix the issue.

**Comment by JohnK1987** [comment]
Look to wiki [configuration via cmake files](https://github.com/mbed-ce/mbed-os/wiki/MbedOS-configuration#configuration-via-cmake-files).

**Pull request closed without merging**